phoenix-core/src/it/java/org/apache/phoenix/end2end/BackwardCompatibilityIT.java

```diff
@@ -437,8 +472,17 @@ public void run() {
     }
 
     // Executes the SQL commands listed in the given operation file from the sql_files directory
-    private void executeQueriesWithCurrentVersion(String operation) throws Exception {
+    private void executeQueriesWithCurrentVersion(String operation,
+            boolean setMaxLookBackAge) throws Exception {
         Properties props = PropertiesUtil.deepCopy(TEST_PROPERTIES);
+        if (setMaxLookBackAge) {
+            props.put(
+                QueryServices.GLOBAL_INDEX_ROW_AGE_THRESHOLD_TO_DELETE_MS_ATTRIB,
+                    Long.toString(0));
+            props.put(CompatBaseScannerRegionObserver.PHOENIX_MAX_LOOKBACK_AGE_CONF_KEY,
+                Integer.toString(15));
```

Comment:
why 15 here?

Reply:
With 15, we are able to reproduce the exception without source code change on this patch. Is it fine? I can update this if there is any better suggestion that I might be unaware of.

Reply:
Any value less than 31 should work for testing purpose. Because during upgrade, SCN is set to be Phoenix version timestamp:
```
    public static final long MIN_SYSTEM_TABLE_TIMESTAMP_4_16_0 = MIN_TABLE_TIMESTAMP + 31;
    public static final long MIN_SYSTEM_TABLE_TIMESTAMP_5_1_0 = MIN_SYSTEM_TABLE_TIMESTAMP_4_16_0;
```

Reply:
can you add explanation comments here so that in the future ppl have a better understanding, thanks